### PR TITLE
created Quiz Performance Table

### PR DESCRIPTION
**Pull request by XI4507**
I have created a Quiz Performance table that provides information about previous quizzes. It displays the performance of student in a particular quiz.

### CODE REVIEW

File: packages/alea-frontend/pages/api/get-quiz-performance.ts
Comment (abchugh) [inline packages/alea-frontend/pages/api/get-quiz-performance.ts:1]: Rename to get-previous-quiz-info

File: packages/api/src/lib/quiz-api.ts [hunk 1]
```diff
@@ -68,3 +69,10 @@ export async function getCourseQuizList(
 ): Promise<QuizStubInfo[]> {
   return (await axios.get(`/api/get-course-quiz-list/${courseId}`)).data;
 }
+
+export async function getQuizPerformance() {
+  const resp = await axios.get(`/api/get-quiz-performance`,{
```
Comment (abchugh) [inline packages/api/src/lib/quiz-api.ts:74]: get-previous-quiz-info

File: packages/alea-frontend/pages/quiz-dash/[courseId].tsx [hunk 2]
```diff
@@ -218,7 +219,10 @@ const QuizDashPage: NextPage = () => {
           quizList={upcomingQuizzes}
           practiceInfo={PRACTICE_QUIZ_INFO[courseId]}
         />
-        <QuizList header={t.previousQuizzes} quizList={previousQuizzes} />
+        <QuizPerofrmanceTable
```
Comment (abchugh) [inline packages/alea-frontend/pages/quiz-dash/[courseId].tsx:222]: typo: QuizPerformanceTable

File: packages/alea-frontend/components/QuizPerformanceTable.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import { PRIMARY_COL } from '@stex-react/utils';
+import dayjs from 'dayjs';
+import Link from 'next/link';
+import { useEffect, useState } from 'react';
+
+function QuizPerofrmanceTable({ quizList, header }) {
+  const [previousQuizData, setPreviousQuizData] =
+    useState<GetPreviousQuizInfoResponse>();
+  useEffect(() => {
+    getQuizPerformance().then(setPreviousQuizData);
```
Comment (abchugh) [inline packages/alea-frontend/components/QuizPerformanceTable.tsx:25]: getPreviousQuizInfo

File: packages/alea-frontend/components/QuizPerformanceTable.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {quizList
+              .sort((a, b) => b.quizStartTs - a.quizStartTs)
+              .map((quiz) => (
+                <TableRow key={quiz.quizId}>
+                  <Link href={`/quiz/${quiz.quizId}`}>
+                    <TableCell sx={{ color: PRIMARY_COL }}>
```
Comment (abchugh) [inline packages/alea-frontend/components/QuizPerformanceTable.tsx:59]: Should this have `cursor: pointer`?

**Review by abchugh — APPROVED** [review]
Looks good. Please submit after renaming files/variables.

File: packages/alea-frontend/components/QuizPerformanceTable.tsx (hunk 4 above)
Comment (XI4507) [reply packages/alea-frontend/components/QuizPerformanceTable.tsx:59]: > Should this have `cursor: pointer`?

without using cursor:pointer it is showing pointer on hovering do i need to explicitly mention cursor: pointer?

**Pull request merged**